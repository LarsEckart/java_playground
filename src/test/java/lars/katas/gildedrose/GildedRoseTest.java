package lars.katas.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

public class GildedRoseTest {

    @Test
    public void foo() throws Exception {

        CombinationApprovals.verifyAllCombinations(this::doUpdateQuality,
                new String[] {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
                new Integer[] {-1, 0, 2, 8, 11},
                new Integer[] {0, 1, 49, 50});
    }

    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[] {new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
