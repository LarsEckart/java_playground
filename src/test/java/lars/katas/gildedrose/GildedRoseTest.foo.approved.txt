[foo, -1, 0] => Item{name='foo', sellIn=-2, quality=0} 
[foo, -1, 1] => Item{name='foo', sellIn=-2, quality=0} 
[foo, -1, 49] => Item{name='foo', sellIn=-2, quality=47} 
[foo, -1, 50] => Item{name='foo', sellIn=-2, quality=48} 
[foo, 0, 0] => Item{name='foo', sellIn=-1, quality=0} 
[foo, 0, 1] => Item{name='foo', sellIn=-1, quality=0} 
[foo, 0, 49] => Item{name='foo', sellIn=-1, quality=47} 
[foo, 0, 50] => Item{name='foo', sellIn=-1, quality=48} 
[foo, 2, 0] => Item{name='foo', sellIn=1, quality=0} 
[foo, 2, 1] => Item{name='foo', sellIn=1, quality=0} 
[foo, 2, 49] => Item{name='foo', sellIn=1, quality=48} 
[foo, 2, 50] => Item{name='foo', sellIn=1, quality=49} 
[foo, 8, 0] => Item{name='foo', sellIn=7, quality=0} 
[foo, 8, 1] => Item{name='foo', sellIn=7, quality=0} 
[foo, 8, 49] => Item{name='foo', sellIn=7, quality=48} 
[foo, 8, 50] => Item{name='foo', sellIn=7, quality=49} 
[foo, 11, 0] => Item{name='foo', sellIn=10, quality=0} 
[foo, 11, 1] => Item{name='foo', sellIn=10, quality=0} 
[foo, 11, 49] => Item{name='foo', sellIn=10, quality=48} 
[foo, 11, 50] => Item{name='foo', sellIn=10, quality=49} 
[Aged Brie, -1, 0] => Item{name='Aged Brie', sellIn=-2, quality=2} 
[Aged Brie, -1, 1] => Item{name='Aged Brie', sellIn=-2, quality=3} 
[Aged Brie, -1, 49] => Item{name='Aged Brie', sellIn=-2, quality=50} 
[Aged Brie, -1, 50] => Item{name='Aged Brie', sellIn=-2, quality=50} 
[Aged Brie, 0, 0] => Item{name='Aged Brie', sellIn=-1, quality=2} 
[Aged Brie, 0, 1] => Item{name='Aged Brie', sellIn=-1, quality=3} 
[Aged Brie, 0, 49] => Item{name='Aged Brie', sellIn=-1, quality=50} 
[Aged Brie, 0, 50] => Item{name='Aged Brie', sellIn=-1, quality=50} 
[Aged Brie, 2, 0] => Item{name='Aged Brie', sellIn=1, quality=1} 
[Aged Brie, 2, 1] => Item{name='Aged Brie', sellIn=1, quality=2} 
[Aged Brie, 2, 49] => Item{name='Aged Brie', sellIn=1, quality=50} 
[Aged Brie, 2, 50] => Item{name='Aged Brie', sellIn=1, quality=50} 
[Aged Brie, 8, 0] => Item{name='Aged Brie', sellIn=7, quality=1} 
[Aged Brie, 8, 1] => Item{name='Aged Brie', sellIn=7, quality=2} 
[Aged Brie, 8, 49] => Item{name='Aged Brie', sellIn=7, quality=50} 
[Aged Brie, 8, 50] => Item{name='Aged Brie', sellIn=7, quality=50} 
[Aged Brie, 11, 0] => Item{name='Aged Brie', sellIn=10, quality=1} 
[Aged Brie, 11, 1] => Item{name='Aged Brie', sellIn=10, quality=2} 
[Aged Brie, 11, 49] => Item{name='Aged Brie', sellIn=10, quality=50} 
[Aged Brie, 11, 50] => Item{name='Aged Brie', sellIn=10, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 2, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 2, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=4} 
[Backstage passes to a TAFKAL80ETC concert, 2, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 2, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 8, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=7, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 8, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=7, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 8, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=7, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 8, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=7, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=1} 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=50} 
[Sulfuras, Hand of Ragnaros, -1, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=0} 
[Sulfuras, Hand of Ragnaros, -1, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=1} 
[Sulfuras, Hand of Ragnaros, -1, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=49} 
[Sulfuras, Hand of Ragnaros, -1, 50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=50} 
[Sulfuras, Hand of Ragnaros, 0, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=0} 
[Sulfuras, Hand of Ragnaros, 0, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=1} 
[Sulfuras, Hand of Ragnaros, 0, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=49} 
[Sulfuras, Hand of Ragnaros, 0, 50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=50} 
[Sulfuras, Hand of Ragnaros, 2, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=0} 
[Sulfuras, Hand of Ragnaros, 2, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=1} 
[Sulfuras, Hand of Ragnaros, 2, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=49} 
[Sulfuras, Hand of Ragnaros, 2, 50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=50} 
[Sulfuras, Hand of Ragnaros, 8, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=8, quality=0} 
[Sulfuras, Hand of Ragnaros, 8, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=8, quality=1} 
[Sulfuras, Hand of Ragnaros, 8, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=8, quality=49} 
[Sulfuras, Hand of Ragnaros, 8, 50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=8, quality=50} 
[Sulfuras, Hand of Ragnaros, 11, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=0} 
[Sulfuras, Hand of Ragnaros, 11, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=1} 
[Sulfuras, Hand of Ragnaros, 11, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=49} 
[Sulfuras, Hand of Ragnaros, 11, 50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=50} 
